Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs

```diff
@@ -14,16 +14,160 @@ static void Main(string[] args)
+
+            for(int i=0; i<equation.Length; i++){
+                if(equation[i]=='*'||equation[i]=='='){
+                    index++;
+                }else{
+                    num[index] += equation[i];
+                }
+            }
+            
+            // now let's find question mark index, to check which number has tha missing digit!
+            // and store the information in the variable questionMarkIndex
+            if(num[0].Contains('?')){
+                questionMarkIndex = 0;
+            }else if(num[1].Contains('?')){
+                questionMarkIndex = 1;
+            }else if(num[2].Contains('?')){
+                questionMarkIndex = 2;
+            }
+
+            if(questionMarkIndex==0){
+                //Console.WriteLine("missing number is First");
+                // Now we retieve the number missing!
+                number[1]= Int32.Parse(num[1]);
+                number[2]= Int32.Parse(num[2]);
+
+                number[0] = number[2]%number[1];
+                if(number[0]!=0){
+                    return -1;
+                }else
+                {
+                    number[0] = number[2]/number[1];
+                }
+                temp = number[0].ToString();
+
+                char[] temporary = temp.ToCharArray();
+                char[] orignal = num[0].ToCharArray();
+            
+                // let's replace ? with the actual result.
+                for(int i=0; i<orignal.Length; i++){
+                    if(orignal[i]=='?'){
+                        orignal[i]=temporary[i];
+                        result= (int)Char.GetNumericValue(temporary[i]); // storing the missing digit in result
+                    }
+                }
+                
+
+                //now let's compare the numbers orignal and temporary
+                num[0] = new String(orignal);
+                temp = new String(temporary);
+                if(String.Equals(num[0],temp)){
+                    return result; // returning the missing digit
+                }else {
+                    return -1;
+                }
+                // Process finished for FIRST operand missing ?
+
+            }else if(questionMarkIndex==1){
+                //Console.WriteLine("missing number is Second");
+                // Now we retieve the number missing!
+                number[0]= Int32.Parse(num[0]);
+                number[2]= Int32.Parse(num[2]);
+
+                number[1] = number[2]%number[0];
+                if(number[1]!=0){
+                    return -1;
+                }else
+                {
+                    number[1] = number[2]/number[0];
+                }
+                temp = number[1].ToString();
+
+                char[] temporary = temp.ToCharArray();
+                char[] orignal = num[1].ToCharArray();
+            
+                // let's replace ? with the actual result.
+                for(int i=0; i<orignal.Length; i++){

```

Comment:
Try exploring `String.IndexOf` and `String.Replace` functions.

Reply:
These were of great help. updated code with these.